docs/source/exporters/onnx/package_reference/configuration.mdx

```diff
@@ -0,0 +1,112 @@
+<!--Copyright 2022 The HuggingFace Team. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# Configuration
+
+Exporting a model to ONNX involves specifying:
+1. The input names
+2. The output names
+3. The dynamic axes: basically which dimensions in the inputs will be dynamic (the other will be considered static)
+4. Dummy inputs used for tracing the model, which is needed in PyTorch to records the graph and convert it to ONNX
+
+This data being model and task dependent, it is represented as configuration classes, each architecture being associated with
+a configuration following the naming convention `ArchitectureNameOnnConfig`. For instance the configuration handling the ONNX
+export of BERT models is `BertOnnxConfig`.
+
+As a lot of the work can be shared accross the different configurations, we chose to have a 3-level class hierarchy:
+1. First, there are the abstract and very generic base classes, handling all the fundamental features while being agnostic to the modality.
+2. Then, there are the middle-end classes, aware of the modality and specifying which input generators should be used for the dummy inputs, but still model-agnostic.
+3. Finally, there are the model-specific classes mentioned above. Those are the ones actually used to export models.
+
+
+## Base classes
+
+[[autodoc]] exporters.onnx.base.OnnxConfig
+
+[[autodoc]] exporters.onnx.base.OnnxConfigWithPast
+
+[[autodoc]] exporters.onnx.base.OnnxSeq2SeqConfigWithPast
+
+
+## Middle-end classes
+
+### Text
+
+[[autodoc]] exporters.onnx.config.TextEncoderOnnxConfig
+
+[[autodoc]] exporters.onnx.config.TextDecoderOnnxConfig
+
+[[autodoc]] exporters.onnx.config.TextSeq2SeqOnnxConfig
+
+
+### Vision
+
+[[autodoc]] exporters.onnx.config.VisionOnnxConfig
+
+
+### Multi-modal
+
+[[autodoc]] exporters.onnx.config.TextAndVisionOnnxConfig
+
+## Supported architectures
```

Comment:
Can we auto-generate this list like we do in `transformers`?

Reply:
I thought of this, maybe in another PR, but it would be perfect yes!